internal/controller/rhdh/backstage.go

```diff
@@ -36,121 +34,166 @@ var ConfigMapNameAndConfigDataKey = map[string]string{
 	AppConfigRHDHDynamicPluginName: "dynamic-plugins.yaml",
 }
 
-func CreateBSSecret(secretName string, secretNamespace, npmRegistry string,
-	ctx context.Context, client client.Client) error {
+func HandleRHDHOperatorInstallation(ctx context.Context, client client.Client, olmClientSet olmclientset.Clientset, namespace string) error {
+	knativeLogger := log.FromContext(ctx)
+
+	// check if subscription exist
+	rhdhSubscription := kubeoperations.CreateSubscriptionObject(
+		BackstageSubscriptionName,
+		namespace,
+		BackstageSubscriptionChannel,
+		"")
+
+	// check if subscription exists
+	subscriptionExists, existingSubscription, err := kubeoperations.CheckSubscriptionExists(ctx, olmClientSet, rhdhSubscription)
+	if err != nil {
+		knativeLogger.Error(err, "Error occurred when checking subscription exists", "SubscriptionName", BackstageSubscriptionName)
+		return err
+	}
+	if !subscriptionExists {
+		if err := kubeoperations.InstallOperatorViaSubscription(ctx, client, olmClientSet, BackstageOperatorGroup, rhdhSubscription); err != nil {
+			knativeLogger.Error(err, "Error occurred when installing operator", "SubscriptionName", BackstageSubscriptionName)
+			return err
+		}
+		knativeLogger.Info("Operator successfully installed", "SubscriptionName", BackstageSubscriptionName)
+	}
+
+	if subscriptionExists {
+		// Compare the current and desired state
+		if !reflect.DeepEqual(existingSubscription.Spec, rhdhSubscription.Spec) {
+			// Set owner reference for proper garbage collection
+			//if err := controllerutil.SetControllerReference(&orchestrator, oslSubscription, r.Scheme); err != nil {
+			//	return err
+			//}
+
+			// Update the existing subscription with the new Spec
+			existingSubscription.Spec = rhdhSubscription.Spec
+			if err := client.Update(ctx, existingSubscription); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func CreateBSSecret(secretNamespace string, ctx context.Context, client client.Client) error {
```

Comment:
Why `CreateBSSecret` and not `CreateRHDHSecret`

Reply:
Renamed to `CreateRHDHSecret`